python/feathub/metric_stores/metric_store.py

```diff
@@ -222,6 +222,27 @@ def _get_metrics_view(
             keep_source_fields=False,
         )
 
+        schema_builder = Schema.new_builder()
+        for feature in derived_feature_view.get_output_features():
+            schema_builder.column(feature.name, feature.dtype)
+
+        return DerivedFeatureView(
+            name=f"{features_desc.name}_periodic_reported_metrics_{window_size_sec}",
+            source=derived_feature_view,
+            features=[
+                Feature(
```

Comment:
However, In the long-term, I think it makes more sense to put the JavaUdfTransform, which is a row-level transformation, in the FeatureView, similar to the filter expr in FeatureView.